code/game/objects/items/stacks/stack.dm

```diff
@@ -265,6 +273,27 @@
 			var/datum/matter_synth/S = synths[i]
 			S.add_charge(charge_costs[i] * extra)
 
+/obj/item/stack/proc/set_amount(var/new_amount, var/no_limits = FALSE)
+	if(new_amount < 0 || new_amount % 1)
+		stack_trace("Tried to set a bad stack amount: [new_amount]")
+		return 0
+	
+	// Clean up the new amount
+	new_amount = max(round(new_amount), 0)
```

Comment:
Why do this when we just exit out if its either negative or decimal anyway?

Reply:
I don't necessarily want to leave the stack_trace forever. It doesn't reaaaally need to be done though.